training-api/src/common.ts

```diff
@@ -10,10 +10,14 @@ export async function api<T, Shape extends ZodRawShape>(
   endpoint: string,
   body: object,
   schema: ZodObject<Shape, any, any, T, any>,
+  cookie?: string,
 ): Promise<T> {
   const resp = await fetch(`${BASE_URL}/${endpoint}`, {
     method: "POST",
-    headers: { "Content-Type": "application/json" },
+    headers: {
+      "Content-Type": "application/json",
+      Cookie: typeof cookie === "string" ? cookie : "",
```

Comment:
Maybe instead of setting this to an empty string, we should leave it unset (to avoid creating issues for the client-side use case where credentials are included by the browser) and do something like this instead:

```
  const options = {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Cookie: undefined,
    },
    body: JSON.stringify(body),
  };
  if (typeof cookie === "string") {
    options.headers.Cookie = cookie;
  }
  const resp = await fetch(`${BASE_URL}/${endpoint}`, options);
  ```

This creates a bunch of typing issues though... I didn't figure out yet how to squash them

Reply:
OK, I verified that whatever we put in the Cookie field of the options object will simply get overwritten by the browser request. So probably we can go ahead with the "token or empty string" approach.